diff --git a/dist/esm/instrumenter/core/instrumenter.js b/dist/esm/instrumenter/core/instrumenter.js
--- a/dist/esm/instrumenter/core/instrumenter.js
+++ b/dist/esm/instrumenter/core/instrumenter.js
@@ -92,7 +92,7 @@
                             {
                                 type: 'select',
                                 name: 'action',
-                                message: `Translate: "${candidate.content}" (${candidate.line}:${candidate.column})?`,
+                                message: `Translate: "${candidate.content}" (${file}:${candidate.line}:${candidate.column})?`,
                                 choices: [
                                     { name: 'Approve', value: 'approve' },
                                     { name: 'Skip', value: 'skip' },
@@ -104,7 +104,7 @@
                         ]);
                         switch (action) {
                             case 'approve':
-                                candidate.key = generateKeyFromContent(candidate.content);
+                                candidate.key ??= generateKeyFromContent(candidate.content);
                                 break;
                             case 'skip':
                                 candidate.skipReason = 'User skipped';
@@ -116,7 +116,7 @@
                                         type: 'input',
                                         name: 'key',
                                         message: 'New key:',
-                                        default: generateKeyFromContent(candidate.content)
+                                        default: candidate.key ?? generateKeyFromContent(candidate.content)
                                     }
                                 ]);
                                 candidate.key = key;
@@ -131,7 +131,7 @@
                                     }
                                 ]);
                                 candidate.content = value;
-                                candidate.key = generateKeyFromContent(value);
+                                candidate.key ??= generateKeyFromContent(value);
                                 break;
                             }
                         }
diff --git a/dist/cjs/instrumenter/core/instrumenter.js b/dist/cjs/instrumenter/core/instrumenter.js
--- a/dist/cjs/instrumenter/core/instrumenter.js
+++ b/dist/cjs/instrumenter/core/instrumenter.js
@@ -98,7 +98,7 @@
                             {
                                 type: 'select',
                                 name: 'action',
-                                message: `Translate: "${candidate.content}" (${candidate.line}:${candidate.column})?`,
+                                message: `Translate: "${candidate.content}" (${file}:${candidate.line}:${candidate.column})?`,
                                 choices: [
                                     { name: 'Approve', value: 'approve' },
                                     { name: 'Skip', value: 'skip' },
@@ -110,7 +110,7 @@
                         ]);
                         switch (action) {
                             case 'approve':
-                                candidate.key = keyGenerator.generateKeyFromContent(candidate.content);
+                                candidate.key ??= keyGenerator.generateKeyFromContent(candidate.content);
                                 break;
                             case 'skip':
                                 candidate.skipReason = 'User skipped';
@@ -122,7 +122,7 @@
                                         type: 'input',
                                         name: 'key',
                                         message: 'New key:',
-                                        default: keyGenerator.generateKeyFromContent(candidate.content)
+                                        default: candidate.key ?? keyGenerator.generateKeyFromContent(candidate.content)
                                     }
                                 ]);
                                 candidate.key = key;
@@ -137,7 +137,7 @@
                                     }
                                 ]);
                                 candidate.content = value;
-                                candidate.key = keyGenerator.generateKeyFromContent(value);
+                                candidate.key ??= keyGenerator.generateKeyFromContent(value);
                                 break;
                             }
                         }
